### ISSUE TITLE
[Windows] ament_clang_tidy reports success when clang-tidy exits with an error

### ISSUE THREAD

**Issue by dskkato** [ament/ament_lint#595]
## Summary

On Windows, `ament_clang_tidy` can report a successful test result even when `clang-tidy.exe` exits with a non-zero status.

This may cause CI to become green although clang-tidy failed.

## Environment

- OS: GitHub Actions `windows-2022`
- ROS 2 distribution: Lyrical
- ROS archive: `release-lyrical-20260807`
- Package under test: [memfd_buffer_backend](https://github.com/dskkato/memfd_buffer_backend/pull/10)
- `clang-tools`: 18.1.8

The related compiler compatibility problem is reported separately in:

- https://github.com/ros2/ros2/issues/1873

## Observed behavior

The `clang-tidy` subprocess fails:

```text
The invocation of "clang-tidy.EXE" failed with error code 1
```

The output contains errors such as:

```
error: error reading 'D:\...\include\memfd_buffer\..': is a directory [clang-diagnostic-error]

C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.44.35207\include\yvals_core.h:908:1:
error: static assertion failed: error STL1000: Unexpected compiler version,
expected Clang 19.0.0 or newer. [clang-diagnostic-error]
```

However, the test is reported as successful:

```
-- run_test.py: return code 0
2/7 Test #2: clang_tidy .......................   Passed
```

The diagnostic parser appears to expect Unix-style paths beginning with `/`.
Therefore, Windows diagnostics using paths such as `C:\...` or `D:\...` may not be recognized. This issue was reported in #594 separately.

More importantly, a non-zero exit status from the clang-tidy subprocess does not appear to be propagated to the final test result.

## Expected behavior

If any `clang-tidy` invocation exits with a non-zero status, the `ament_clang_tidy` test should fail, regardless of whether the diagnostic
parser recognizes the emitted paths.

The diagnostic parser should also support Windows drive-letter paths and backslash-separated paths.

## Reproduction

The complete GitHub Actions log is available here:

https://github.com/dskkato/memfd_buffer_backend/actions/runs/34746595256

The relevant workflow is:

https://github.com/dskkato/memfd_buffer_backend/blob/56921157e56871bc3d37493ff30677569b36914f/.github/workflows/ros2-rolling.yml
